File: ytrue-admin-server-csharp/Ytrue.Infrastructures/Response/ServerResponseInfo.cs
using Ytrue.Infrastructures.Extensions;

namespace Ytrue.Infrastructures.Response;

public sealed class ServerResponseInfo : IServerResponseInfo
{
    private ServerResponseInfo(string code, string message)
    {
        _code = code;
        _message = message;
    }

    private readonly string _code;

    private readonly string _message;

    
    /// <summary>
    /// 错误返回
    /// </summary>
    /// <param name="message"></param>
    /// <returns></returns>
    public static IServerResponseInfo Error(string message)
    {
        return new ServerResponseInfo(ServerResponseEnum.InternalServerError.GetDescription(), message);
    }


    public string Code()
    {
        return _code;
    }

    public string Message()
    {
        return _message;
    }
}

File: ytrue-admin-server-java/ytrue-infra/ytrue-infra-core/src/main/java/com/ytrue/infra/core/response/ServerResponseInfo.java
package com.ytrue.infra.core.response;

import com.ytrue.infra.core.base.IServerResponseInfo;

/**
 * ServerResponseInfo 类用于封装服务器响应的状态码和消息信息。
 *
 * @param code    状态码
 * @param message 信息
 */
public record ServerResponseInfo(String code, String message) implements IServerResponseInfo {

    /**
     * 创建一个包含错误消息的响应信息，使用默认的失败状态码。
     *
     * @param message 错误消息
     * @return IServerResponseInfo 的实例
     */
    public static IServerResponseInfo error(String message) {
        return new ServerResponseInfo(ServerResponseInfoEnum.CUSTOM_FAIL_MESSAGE.code(), message);
    }

    /**
     * 创建一个包含自定义状态码和消息的响应信息。
     *
     * @param code    自定义状态码
     * @param message 自定义错误消息
     * @return IServerResponseInfo 的实例
     */
    public static IServerResponseInfo error(String code, String message) {
        return new ServerResponseInfo(code, message);
    }
}
